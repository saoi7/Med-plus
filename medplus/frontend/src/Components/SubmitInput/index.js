import React from 'react';

function SubmitInput(props) {
    return (
        <div className="add-med-input font-small">
            <input type="submit" value={props.labelText} className="hidden-submit-button background-grey font-small"></input>
        </div>
    );
}

export default SubmitInput;
