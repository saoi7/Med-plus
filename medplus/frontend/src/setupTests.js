// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { unmountComponentAtNode } from "react-dom";

let test_dom_container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  test_dom_container = document.createElement("div");
  document.body.appendChild(test_dom_container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(test_dom_container);
  test_dom_container.remove();
  test_dom_container = null;
});

export { test_dom_container };
