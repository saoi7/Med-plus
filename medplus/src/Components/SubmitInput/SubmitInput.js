import React from 'react';

function SubmitInput(props) {
    return (
        <div className="add-med-input font-medium">
            <input type="submit" value={props.labelText} className="hidden-submit-button font-medium"></input>
        </div>
    );
}

export { SubmitInput };