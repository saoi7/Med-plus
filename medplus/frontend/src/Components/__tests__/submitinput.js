import SubmitInput from '../SubmitInput';
import {test_dom_container} from '../../setupTests';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';

test("check no unintended changes have been made to rendered component", () => {
    const component= renderer.create(
        <SubmitInput />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("check html elements of SubmitInput component", () => {
    act(() => {
        render(<SubmitInput />, test_dom_container);
    });

    const ELEMENT = test_dom_container.childNodes;
    expect(ELEMENT[0].tagName).toBe('DIV');

    const CHILDREN = ELEMENT[0].childNodes;
    expect(CHILDREN.length).toBe(1);

    const INPUT_ELEMENT = CHILDREN[0];
    expect(INPUT_ELEMENT.tagName).toBe('INPUT');
});

test("check SubmitInput text content", () => {
    const TEXT_PROP = 'abcdefg';
    act(() => {
        render(<SubmitInput labelText={TEXT_PROP} />, test_dom_container);
    });

    const ELEMENT = test_dom_container.childNodes;

    const INPUT_ELEMENT = ELEMENT[0].childNodes[0];
    expect(INPUT_ELEMENT.value).toBe(TEXT_PROP);
});
