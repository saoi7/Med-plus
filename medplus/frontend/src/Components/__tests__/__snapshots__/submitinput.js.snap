// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check no unintended changes have been made to rendered component 1`] = `
<div
  className="add-med-input font-small"
>
  <input
    className="hidden-submit-button font-small"
    type="submit"
  />
</div>
`;
